app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments, through => :posts
```

Comment:
User has many comments,  shouldn't be through post, you already have your comment model and table, it can stand alone